godot_project/editor/input_handlers/molecular_structures/add_posing_atom_input_handler.gd

```diff
@@ -240,13 +241,13 @@ func _update_candidates() -> void:
 			for atom_id: int in atomic_structure.get_valid_atoms():
 				var atom_position: Vector3 = atomic_structure.atom_get_position(atom_id)
 				_atom_grid.add_item(atom_position, atom_id)
-	var index: int = 0
-	while index < _candidates.size():
+	var index: int = _candidates.size() - 1
+	while index >= 0:
 		var candidate: AtomCandidate = _candidates[index]
 		if _atom_grid.has_any_closer_than(candidate.atom_position, MIN_DISTANCE_TO_ATOMS):
 			_candidates.remove_at(index)
 		else:
-			index += 1
+			index -= 1
```

Comment:
I noticed this loop would fail if more than 1 overlaping was detected, because the index of _candidates would shift with every match